web/server/codechecker_server/api/report_server.py

```diff
@@ -623,11 +623,26 @@ def get_report_details(session, report_ids):
         extended_data.filePath = file_path
         extended_data_list[report_id].append(extended_data)
 
+    # Get Comments for report data
+    comment_data_list = defaultdict(list)
+    comment_query = session.query(Comment, Report.id, File.filepath)\
```

Comment:
We need to filter the comments by the `report_ids` function parameter:
```py
.filter(Report.id.in_(report_ids)) \
```
This way we do not query everything from the database which is not used.